Server/mods/deathmatch/logic/lua/CLuaMain.cpp

```diff
@@ -658,3 +660,191 @@ int CLuaMain::OnUndump(const char* p, size_t n)
+//
+// CLuaMain::LoadLuaLib
+//
+// Load a Lua lib of a given name
+//
+///////////////////////////////////////////////////////////////
+bool CLuaMain::LoadLuaLib(lua_State* L, SString strName, SString& strError)
+{
+    SString strPath = strName;
+    // Name format shouldn't include slashes.  Subdirs are dots.
+    ReplaceOccurrencesInString(strPath, "\\", "");
+    ReplaceOccurrencesInString(strPath, "/", "");
+    ReplaceOccurrencesInString(strPath, ".", "/");
+
+    SString strResPath = m_pResource->IsResourceZip() ? m_pResource->GetResourceCacheDirectoryPath() : m_pResource->GetResourceDirectoryPath();
+
+    std::vector<char> buffer;
+    // Try <resource>/?.lua
+    SString strFilePath = PathJoin(strResPath, strPath + ".lua");
+    if (FileExists(strFilePath))
+        FileLoad(strFilePath, buffer);
+    else
+    {
+        // Don't use a format string for safety, so we construct the error by hand
+        strError += "error loading module " + strName + " from file " + strFilePath + ":\n\t The specified module could not be found";
+
+        // Try <resource>/?/init.lua
+        strFilePath = PathJoin(strResPath, strPath, "init.lua");
+        if (FileExists(strFilePath))
+            FileLoad(strFilePath, buffer);
+        else
+        {
+            strError += "\n";
+            strError += "error loading module " + strName + " from file " + strFilePath + ":\n\t The specified module could not be found";
+            return false;
+        }
+    }
+
+    if (buffer.size() > 0)
+    {
+        int luaSavedTop = lua_gettop(L);
+        LoadScriptFromBuffer(&buffer.at(0), buffer.size(), strFilePath.c_str(), false);
+        // Only keep the first return value
+        if (lua_gettop(L) > luaSavedTop)
+            lua_settop(L, luaSavedTop + 1);
+
+        if (lua_type(L, -1) == LUA_TNIL)
+        {
+            strError += "error loading module " + strName + " from file " + strFilePath + ":\n\t Module didn't return a value";
+            return false;
+        }
+
+        SetPackage(L, strName);            // Store our package into package.loaded
+        return true;
+    }
+    else
+        strError += "error loading module " + strName + " from file " + strFilePath + ":\n\t Error loading script file";
+
+    return false;
+}
+
+///////////////////////////////////////////////////////////////
+//
+// CLuaMain::LoadClib
+//
+// Load a C lib of a given name
+//
+///////////////////////////////////////////////////////////////
+bool CLuaMain::LoadClib(lua_State* L, SString strName, SString& strError)
+{
+    SString strPath = strName;
+    // Name format shouldn't include slashes.  Subdirs are dots.
+    ReplaceOccurrencesInString(strPath, "\\", "");
+    ReplaceOccurrencesInString(strPath, "/", "");
+    ReplaceOccurrencesInString(strPath, ".", "/");
+
+    strPath = PathJoin(g_pServerInterface->GetModManager()->GetServerPath(), SERVER_BIN_PATH_MOD, "modules", strPath);
+
+#ifdef WIN32
+    strPath += ".dll";
```

Comment:
or `.lua`

Reply:
No, `.lua` is available in `LoadLuaLib`

Reply:
i just tested, and lua script with .dll extension not working

Reply:
What do you mean? There is a separate function, `LoadLuaLib`, for loading .lua files.  That is already working.  This function is for loading C extensions.